src/main/kotlin/app/urbanflo/urbanflosumoserver/model/network/SumoEdge.kt

```diff
@@ -18,5 +18,15 @@ data class SumoEdge(
     @field:JacksonXmlProperty(isAttribute = true)
     val width: Double,
     @field:JacksonXmlProperty(isAttribute = true)
-    val name: String?
-)
\ No newline at end of file
+    val name: String?,
+    @field:JacksonXmlProperty(isAttribute = true)
+    val spreadType: String?,
+) {
+    init {
+        spreadType?.let {
+            require(it.lowercase() in listOf("right", "center", "roadCenter")) {
```

Comment:
Wouldn't this not work with `roadCenter` since the type is always lowercased?

Reply:
Good point, I'll fix that rn